frontend/src/console/actions.ts

```diff
@@ -70,3 +111,57 @@ export const listUserInstances = (): OcimThunkAction<void> => async dispatch =>
       });
     });
 };
+
+export const updateFieldValue = (
+  instanceId: number,
+  fieldName: string,
+  value: string
+): OcimThunkAction<void> => async dispatch => {
+  dispatch({
+    type: Types.UPDATE_INSTANCE_INFO,
+    fieldName
+  });
+
+  setTimeout(function() {
+    dispatch({
+      type: Types.UPDATE_INSTANCE_INFO_SUCCESS,
+      data: {
+        [fieldName]: value
+      }
+    });
+    // dispatch({
+    //   type: Types.UPDATE_INSTANCE_INFO_FAILURE,
+    //   data: {
+    //     [fieldName]: value
+    //   }
+    // });
+  }, 2000);
```

Comment:
You can also write this using `await` which lets you include the error handling code uncommented:

```typescript
try {
    await new Promise((resolve) => setTimeout(resolve, 2000)); // sleep for 2 seconds
    dispatch({
        type: Types.UPDATE_INSTANCE_INFO_SUCCESS,
        data: {
          [fieldName]: value
        }
    });
} catch {
    dispatch({
      type: Types.UPDATE_INSTANCE_INFO_FAILURE,
      data: {
        [fieldName]: value
      }
    });
}
```